### PR TITLE
Add UMD wrapper

### LINKED ISSUE #515: Make d3fc work as commonjs package
I am trying to make d3fc to work within my React application, but d3fc.js doesn't have module.exports.

I tried adding
```
var computeLayout = require('css-layout');
//code
module.exports = fc;
````

But I am getting an error 

```
Uncaught TypeError: Cannot read property 'width' of undefinedapplyLayout @ d3fc.js:1900(anonymous function) @ d3fc.js:1929(anonymous function) @ d3.js:962d3_selection_each @ d3.js:968d3_selectionPrototype.each @ d3.js:961fc.layout.layout @ d3fc.js:1915d3_selectionPrototype.call @ d3.js:975d3.selection.layout @ d3fc.js:1843(anonymous function) @ d3fc.js:590(anonymous function) @ d3.js:962d3_selection_each @ d3.js:968d3_selectionPrototype.each @ d3.js:961fc.chart.linearTimeSeries.linearTimeSeries @ d3fc.js:537d3_selectionPrototype.call @ d3.js:975React.createClass.reRender @ StocksContainer.jsx:71React.createClass._onChange @ StocksContainer.jsx:34EventEmitter.emit @ events.js:74_.extend.emitChange @ IndexStore.js:24(anonymous function) @ IndexStore.js:55Dispatcher.$Dispatcher_invokeCallback @ Dispatcher.js:220Dispatcher.dispatch @ Dispatcher.js:195assign.handleViewAction @ AppDispatcher.js:12IndexActions.loadIndexInfo @ IndexActions.js:8(anonymous function) @ IndexAPI.js:8jQuery.Callbacks.fire @ jquery.js:3099jQuery.Callbacks.self.fireWith @ jquery.js:3211done @ jquery.js:8264jQuery.ajaxTransport.send.callback @ jquery.js:8605
```

### PR DESCRIPTION
**Pull request by chrisprice**
Tested with globals (default behavior), with browserify (both referencing the `d3fc.js` file and as a package) but not with AMD (religious preference).

Fixes #515.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -278,7 +282,7 @@ module.exports = function(grunt) {
                 helpers: 'tests/beforeEachSpec.js'
             },
             test: {
-                src: ['dist/*.js'],
+                src: ['dist/d3fc.min.js'],
```
Comment (chrisprice) [inline Gruntfile.js]: Not strictly necessary for this change but stops the minified and non-minified code being included (which is just weird).

**Comment by ColinEberhardt** [comment]
Looks good to me :-)

I guess the boilerplate UMD was based on one of the following:

https://github.com/umdjs/umd?files=1

If so, would be useful to reference the file it was based on. In css-layout I found a cut and paste error when someone had added the UMD code, so knowing what it was based on does help!

**Comment by chrisprice** [comment]
Fair point, added.

**Comment by ColinEberhardt** [comment]
It's a shame it has to be added like this, would be better as a grunt task which takes a parameterised UMD wrapper and applies it. Anyhow, happy to see this merged :+1:

**Comment by ColinEberhardt** [comment]
It already a exists ... of course!

https://github.com/bebraw/grunt-umd

**Comment by chrisprice** [comment]
Another good shout. I've still had to make a few small changes to `src/fc.js` to get it to work but otherwise it's a lot cleaner. In terms of packaging, have you seen that d3 is changing it's approach e.g. https://github.com/d3/d3-selection?

**Comment by ColinEberhardt** [comment]
That looks much better :-)

Ive seen that d3 is being broken up into multiple packages, makes sense considering how diverse it's functionality is.

**Pull request merged**